### PR TITLE
docs: setup jupyterbook and  add .readthedocs.yml

### PR DESCRIPTION
**Pull request by anujsinha3**
#4 #11 

Setup JupyterBook / ReadTheDocs

- [x]  add _config.yml file
- [x]  add _toc.yml file
- [x]  add initial intro.md
- [x]  add .readthedocs.yml

### CODE REVIEW

**Comment by anujsinha3** [comment]
I see the ci pipelines have also run for this PR.

**Comment by anujsinha3** [comment]
Also, I have kept the `references.bib` so that whenever MAW team members start populating the docs, they will have a template for adding references.

**Comment by anujsinha3** [comment]
Sample build: https://mawpy-as.readthedocs.io/en/latest/intro.html

File: docs/requirements.txt [hunk 1]
```diff
@@ -0,0 +1,3 @@
+jupyter-book
+matplotlib
+numpy
```
Comment (lsetiawan) [inline docs/requirements.txt:3]: Please add these to the `pyproject.toml` instead as an optional 'docs' dependencies.

File: .readthedocs.yml [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+sphinx:
+  configuration: docs/conf.py
+
+# Optionally build your docs in additional formats such as PDF and ePub
+formats: []
+
+# Optionally set the version of Python and requirements required to build your docs
+python:
+  install:
+    - requirements: docs/requirements.txt # Path to your requirements.txt file
```
Comment (lsetiawan) [inline .readthedocs.yml:28]: This should use `pip` install method like it was

**Review by carlosgjs — APPROVED** [review]
LGTM other than Don's comments

File: docs/requirements.txt (hunk 1 above)
Comment (anujsinha3) [reply docs/requirements.txt]: Updated as per comment.

File: .readthedocs.yml (hunk 2 above)
Comment (anujsinha3) [reply .readthedocs.yml]: updated as per comment.  
Sample build: https://mawpy-as.readthedocs.io/en/latest/intro.html

**Pull request merged**